extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -1872,6 +1871,44 @@ private static QueryDefinition makeQueryDefinition(
       } else {
         return queryDef;
       }
+    } else if (querySpec.getDestination() instanceof ExportMSQDestination) {
+      final ExportMSQDestination exportMSQDestination = (ExportMSQDestination) querySpec.getDestination();
+      final StorageConnectorProvider storageConnectorProvider = exportMSQDestination.getStorageConnectorProvider();
+
+      final ResultFormat resultFormat = exportMSQDestination.getResultFormat();
+
+      // If the statement is a 'REPLACE' statement, delete the existing files at the destination.
+      if (exportMSQDestination.getReplaceTimeChunks() != null) {
+        if (Intervals.ONLY_ETERNITY.equals(exportMSQDestination.getReplaceTimeChunks())) {
+          StorageConnector storageConnector = storageConnectorProvider.get();
+          try {
+            storageConnector.deleteRecursively("");
```

Comment:
This worries me because it means that if someone provides a wrong path by accident, we'll delete a bunch of their stuff. It would be better to only delete files that we can confirm came from a previous Druid export. To achieve that we could write our files in a special directory structure, or we could include a manifest file and check for the prior manifest.

The manifest file is probably a good idea anyway, since it makes it easier for readers to be sure they got the entire set of files.

If we use the Hive "symlink manifest" format in particular, then engines like Presto and Spark will be able to more easily read our exports.

Anyway, whatever we do here, I think we should do something different than blindly deleting everything in the path the user provides. It's too likely that it will lead to accidental data deletion.

Reply:
Even I think a manifest file is a better alternative. Since it would be a bit more work, what we can do now is to remove the delete call, document it as a known bug and get this in druid 29. 
Post that, we can add the manifest deletion as part of a follow up PR. 
How does that plan sound to you @gianm @adarshsanjeev  ?

Reply:
Instead of remove the delete call, we can add a directory like `druid-export/` to the path, and delete it for REPLACE. This way, we would not delete any other files unless the user stores them in a directory with the same name.